src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -405,6 +456,24 @@ InvokeHostFunctionOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx)
     return true;
 }
 
+bool
+InvokeHostFunctionOpFrame::doCheckValid(SorobanNetworkConfig const& config,
+                                        uint32_t ledgerVersion)
+{
+    // check wasm size if uploading contract
+    for (auto const& hostFn : mInvokeHostFunction.functions)
+    {
+        auto const& args = hostFn.args;
+        if (args.type() == HOST_FUNCTION_TYPE_UPLOAD_CONTRACT_WASM &&
+            args.uploadContractWasm().code.size() >
+                config.maxContractSizeBytes())
+        {
+            return false;
+        }
+    }
+    return doCheckValid(ledgerVersion);
+}
+
 bool
 InvokeHostFunctionOpFrame::doCheckValid(uint32_t ledgerVersion)
```

Comment:
It's safer to throw here and never call it to make sure we're not using it directly by accident.

Reply:
Good point. Fixed